packages/block-editor/src/components/spacing-sizes-control/style.scss

```
@@ -21,14 +33,16 @@
 
 	.components-range-control__mark {
 		height: $grid-unit-05;
```

It’s quite minute (especially now with their reduced width), but I believe the base positioning of the marks don’t account for their width so this should place them more accurately.
```suggestion
	.components-range-control__mark {
		transform: translateX(-50%);
		height: $grid-unit-05;
```